src/issue-strategies/bug-maintenance.js

```diff
@@ -1,7 +1,18 @@
+import * as jiraOperations from '../jira-operations.js';
+
 export function apply(issue, jiraClientAPI) {
   if(issue === null || issue.fields.status.statusCategory.colorName !== 'yellow') {
     return Promise.reject(new Error(`Issue ${issue.key} is not open to commit against`));
   }
 
+  if(issue.fields.issuelinks){
+    return jiraOperations.findParent(issue, jiraClientAPI).then((parent)=>{
+      if(!parent){
+        return Promise.resolve(true);
+      }
+      return parent.fields.status.statusCategory.colorName === 'yellow' ? Promise.resolve(true) : Promise.reject(new Error(`Cannot commit issue ${issue.key} because parent issue ${parent.key} is not available to commit against.`));
+    });
+  }
+
   return Promise.resolve(true);
```

Comment:
We can simplify this greatly. I imagine `findParent` should know what to do if `issue.fields.issuelinks` is falsy. These two blocks translate the same way to me:
```js
if (!parent) {
  return Promise.resolve(true);
}

...

if (!issue.fields.issuelinks) {
  return Promise.resolve(true);
}
```
Further simplification can be done with the following:
```js
if(!parent || parent.fields.status.statusCategory.colorName === 'yellow') {
  return Promise.resolve(true);
}
return Promise.reject(new Error(/* error message */));
```